### PR TITLE
[spv-out] Add missing math functions

### PR DESCRIPTION
**Pull request by lain-dono**
Implementation of `sinh`, `cosh`, `tanh`, `mod`, `frexp`, `ldexp`, `faceForward`, `exp`, `exp2`, `log`, `log2`, `sign`, `fma`, `mix`, `step`, `smoothstep`, `sqrt`, `determintant`, `reflect`.

At least `mod`, `frexp`, `ldexp`, `determintant` require some checking.

Perhaps the [specification](https://gpuweb.github.io/gpuweb/wgsl.html) missed the `refract` function.

### CODE REVIEW

File: src/back/spv/writer.rs [hunk 1]
```diff
@@ -1680,8 +1686,43 @@ impl Writer {
...
+                        MathOp::Single(op)
+                    }
+                    Mf::Fma => MathOp::Triple(spirv::GLOp::Fma),
+                    Mf::Mix => {
+                        let op = match self
+                            .get_type_inner(&ir_module.types, arg0_lookup_ty)
+                            .scalar_kind()
+                        {
+                            Some(crate::ScalarKind::Float) => spirv::GLOp::FMix,
+                            Some(crate::ScalarKind::Sint) => spirv::GLOp::IMix,
```
Comment (kvark) [inline src/back/spv/writer.rs:1716]: There is a problem here. Looks like `IMix` is not in the extended instruction set - https://www.khronos.org/registry/spir-v/specs/1.0/GLSL.std.450.html
So for integers we'll have to actually encode the arithmetic expressions in here

File: src/back/spv/writer.rs [hunk 2]
```diff
@@ -1707,7 +1748,31 @@ impl Writer {
...
+                            self.gl450_ext_inst_id,
+                            spirv::GLOp::Determinant,
+                            result_type_id,
+                            id,
+                            &[arg0_id],
+                        );
+                        MathOp::Other(inst, result_lookup_ty)
+                    }
+
+                    Mf::Outer | Mf::ReverseBits | Mf::CountOneBits => {
```
Comment (kvark) [inline src/back/spv/writer.rs:1775]: amazing, love the exhaustive enums!

**Review by kvark — COMMENTED** [review]
Looks great, just one small concern here.

File: src/back/spv/writer.rs (hunk 1 above)
Comment (lain-dono) [reply src/back/spv/writer.rs]: Done

**Review by kvark — APPROVED** [review]
Thank you!

**Pull request merged**